lib/Graph.pm

```diff
@@ -1825,6 +1825,28 @@ sub directed_copy {
 
 *directed_copy_graph = \&directed_copy;
 
+sub is_bipartite {
+    my $g = shift;
```

Comment:
Generally I avoid using `shift`, since that breaks idioms like `&expect_undirected` you can see a few lines above. Also, does this have any requirements like only working with directed graphs? If so, please put those assertions in.

Reply:
I note that other DFS-using methods do similar to this, so it might be that no changes are needed.

Reply:
You are right, most of the subroutines in `Graph.pm` leaves `@_` untouched, I will adjust.